internal/daemon/daemon.go

```diff
@@ -356,13 +360,13 @@ func logit(handler http.Handler) http.Handler {
 func (d *Daemon) Init() error {
 	listenerMap := make(map[string]net.Listener)
 
-	if listener, err := getListener(d.normalSocketPath, listenerMap); err == nil {
+	if listener, err := getListener(d.normalSocketPath, listenerMap, d.dry); err == nil {
```

Comment:
I could be completely wrong here, but looking at the changes here for getListener makes me uncomfortable. I would have classed ```net.Dail()``` as a system state changing action, which is outside the scope of a dry run. In fact, it feels like a very slippery slope here, trying to plumb in new code paths for dry running at this level. For example, getListener() now has a completely new return state ```nil, nil`` that did not exist before, and which requires tests and subsequent code changes to accommodate this. Again, I stand to be corrected.

Reply:
@anpep I feel there is a danger I lost sync on where this is going, so I will catch up with you when you are back before I continue my review. It feels like I may be missing something.